src/Symfony/Bridge/PhpUnit/Legacy/ForwardCompatTestTraitForV5.php

```diff
@@ -215,6 +215,51 @@ public static function assertIsIterable($actual, $message = '')
         static::assertInternalType('iterable', $actual, $message);
     }
 
+    /**
+     * @param string $message
+     *
+     * @return void
+     */
+    public static function assertFinite($actual, $message = '')
+    {
+        if (method_exists(TestCase::class, 'assertFinite')) {
+            parent::assertFinite($actual, $message);
+
+            return;
+        }
+        static::assertTrue(is_finite($actual), $message ? $message : "Failed asserting that $actual is finite.");
```

Comment:
are we sure the blind cast of `$actual` to string is what phpunit does? this will trigger a warning, and prevent correct reporting, no?

Reply:
Current implementation of PhpUnit does the same thing (with many calls).
Here is a "summary" of PhpUnit

```php
class Assert
{
    public static function assertFinite($actual, $message = '')
    {
        static::assertThat($actual, static::isFinite(), $message);
    }

    public static function isFinite()
    {
        return new IsFinite;
    }

   public static function assertThat($value, Constraint $constraint, $message = '')
   {
        // ...
        $constraint->evaluate($value, $message);
    }
}

class IsFinite
{
    public function evaluate($other, $description = '', $returnResult = false)
    {
        // ...
        if ($this->matches($other)) {
            // ...
        }
        // ...
    }
    protected function matches($other)
    {
        return \is_finite($other);
    }
}
```

Reply:
Sorry, I meant the message: `"Failed asserting that $actual is finite."`
phpunit uses `$this->exporter()->export($other)` in the `Constraint` class instead of putting the value directly.

Reply:
Solved by adding `static::assertInternalType('float', $actual, $message);` before.